framework/decode/vulkan_replay_consumer_base.cpp

```diff
@@ -4155,23 +4155,32 @@ VulkanReplayConsumerBase::OverrideCreateBuffer(PFN_vkCreateBuffer
 
     VkResult                              result = VK_SUCCESS;
     VulkanResourceAllocator::ResourceData allocator_data;
-    auto                                  replay_buffer      = pBuffer->GetHandlePointer();
-    auto                                  capture_id         = (*pBuffer->GetPointer());
-    auto                                  replay_create_info = pCreateInfo->GetPointer();
-
+    auto                                  replay_buffer               = pBuffer->GetHandlePointer();
+    auto                                  capture_id                  = (*pBuffer->GetPointer());
+    auto                                  replay_create_info          = pCreateInfo->GetPointer();
+    VkBufferCreateInfo                    modified_create_info        = *replay_create_info;
+    uint32_t                              modified_queue_family_index = 0;
     // Check for a buffer device address.
     bool                uses_address         = false;
     VkBufferCreateFlags address_create_flags = 0;
     VkBufferUsageFlags  address_usage_flags  = 0;
+
+    if ((modified_create_info.sharingMode == VK_SHARING_MODE_CONCURRENT) &&
+        (modified_create_info.queueFamilyIndexCount > 0) && (modified_create_info.pQueueFamilyIndices != nullptr))
+    {
+        modified_create_info.queueFamilyIndexCount = 1;
```

Comment:
I don't think we can assert that all buffers with concurrent usage only use the first queue family. This is a lie for traces in which buffers are used by multiple families and violates the Vulkan specification. 
https://registry.khronos.org/vulkan/specs/1.3-extensions/html/vkspec.html#VkBufferCreateInfo